include/crocoddyl/multibody/contacts/contact-1d.hxx

```diff
@@ -0,0 +1,157 @@
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+
+template <typename Scalar>
+ContactModel1DTpl<Scalar>::ContactModel1DTpl(boost::shared_ptr<StateMultibody> state,
+                                             const FrameTranslationTpl<Scalar>& xref, const std::size_t nu,
+                                             const Vector2s& gains)
+    : Base(state, 1, nu), xref_(Scalar(xref.translation[2])), gains_(gains) {
+  id_ = xref.id;
+  std::cerr << "Deprecated: Use constructor which is not based on FrameTranslation." << std::endl;
+}
+
+template <typename Scalar>
+ContactModel1DTpl<Scalar>::ContactModel1DTpl(boost::shared_ptr<StateMultibody> state,
+                                             const FrameTranslationTpl<Scalar>& xref, const Vector2s& gains)
+    : Base(state, 1), xref_(Scalar(xref.translation[2])), gains_(gains) {
+  id_ = xref.id;
+  std::cerr << "Deprecated: Use constructor which is not based on FrameTranslation." << std::endl;
+}
+
+#pragma GCC diagnostic pop
+
+template <typename Scalar>
+ContactModel1DTpl<Scalar>::~ContactModel1DTpl() {}
+
+template <typename Scalar>
+void ContactModel1DTpl<Scalar>::calc(const boost::shared_ptr<ContactDataAbstract>& data,
+                                     const Eigen::Ref<const VectorXs>&) {
+  Data* d = static_cast<Data*>(data.get());
+  pinocchio::updateFramePlacement(*state_->get_pinocchio().get(), *d->pinocchio, id_);
+  pinocchio::getFrameJacobian(*state_->get_pinocchio().get(), *d->pinocchio, id_, pinocchio::LOCAL, d->fJf);
+  d->v = pinocchio::getFrameVelocity(*state_->get_pinocchio().get(), *d->pinocchio, id_);
+  d->a = pinocchio::getFrameAcceleration(*state_->get_pinocchio().get(), *d->pinocchio, id_);
+
+  d->Jc.row(0) = d->fJf.row(2);
+
+  d->vw = d->v.angular();
+  d->vv = d->v.linear();
+
+  d->a0[0] = d->a.linear()[2] + d->vw[0] * d->vv[1] - d->vw[1] * d->vv[0];
+
+  if (gains_[0] != 0.) {
+    d->a0[0] += gains_[0] * (d->pinocchio->oMf[id_].translation()[2] - xref_);
+  }
+  if (gains_[1] != 0.) {
+    d->a0[0] += gains_[1] * d->vv[2];
+  }
+}
+
+template <typename Scalar>
+void ContactModel1DTpl<Scalar>::calcDiff(const boost::shared_ptr<ContactDataAbstract>& data,
+                                         const Eigen::Ref<const VectorXs>&) {
+  Data* d = static_cast<Data*>(data.get());
+  const pinocchio::JointIndex joint = state_->get_pinocchio()->frames[d->frame].parent;
+  pinocchio::getJointAccelerationDerivatives(*state_->get_pinocchio().get(), *d->pinocchio, joint, pinocchio::LOCAL,
+                                             d->v_partial_dq, d->a_partial_dq, d->a_partial_dv, d->a_partial_da);
+  const std::size_t nv = state_->get_nv();
+  pinocchio::skew(d->vv, d->vv_skew);
+  pinocchio::skew(d->vw, d->vw_skew);
+  d->fXjdv_dq.noalias() = d->fXj * d->v_partial_dq;
+  d->fXjda_dq.noalias() = d->fXj * d->a_partial_dq;
+  d->fXjda_dv.noalias() = d->fXj * d->a_partial_dv;
+
+  d->da0_dx.leftCols(nv).row(0) = d->fXjda_dq.row(2);
```

Comment:
Same here, it is not compatible with codegen.
See below for details on how to fix it.